src/components/search/components/stack.tsx

```diff
@@ -6,63 +6,62 @@ import {loadCourse} from 'lib/courses'
 import {loadLesson} from 'lib/lessons'
 import {loadPlaylist} from 'lib/playlists'
 
+type StackResource = {slug: string; type: string}
+
 type StackProps = {
-  resources: any[]
+  resources: StackResource[]
 }
 
-const Stack: FunctionComponent<StackProps> = ({resources}) => {
-  let res: any[] = []
-  resources.forEach((resource) => {
-    if (resource.type === 'lesson') {
-      const {data: lesson} = useSWR(resource.slug, loadLesson)
```

Comment:
@vojtaholik this was a prime example of a conditional hook use. Instead, I broke the element out into individual components.

Reply:
Noted!